code/modules/plumbing/ducts.dm

```diff
@@ -101,7 +106,7 @@ All the important duct code:
 ///connect to a duct
 /obj/machinery/duct/proc/connect_duct(obj/machinery/duct/D, direction, ignore_color)
 	var/opposite_dir = turn(direction, 180)
-	if(!active || !D.active)
+	if(!active || !D.active || (D in SSfluids.build_queued)) //if they're already build queued, someone else primed them already and we'll end up being connected anyway
```

Comment:
see this will be suffering if build_queued is a long list

Reply:
I’m gonna see if this is still needed